pkg/engine/lifecycletest/pulumi_test.go

```diff
@@ -5550,204 +5550,168 @@ func TestContinueOnError(t *testing.T) {
 	assert.Equal(t, resource.URN("urn:pulumi:test::test::pkgB:m:typB::failing"), snap.Resources[3].URN)
 }
 
-// TestStackOutputsProgramError tests that previous stack outputs aren't deleted when an update fails because
-// of a program error.
-func TestStackOutputsProgramError(t *testing.T) {
+func TestUpContinueOnErrorCreate(t *testing.T) {
 	t.Parallel()
 
-	var step int
-
 	loaders := []*deploytest.ProviderLoader{
 		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
 			return &deploytest.Provider{}, nil
-		}),
+		}, deploytest.WithoutGrpc),
+		deploytest.NewProviderLoader("pkgB", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{
+				CreateF: func(urn resource.URN, news resource.PropertyMap, timeout float64,
+					preview bool,
+				) (resource.ID, resource.PropertyMap, resource.Status, error) {
+					return "", nil, resource.StatusOK, errors.New("intentionally failed create")
+				},
+			}, nil
+		}, deploytest.WithoutGrpc),
 	}
 
 	programF := deploytest.NewLanguageRuntimeF(func(_ plugin.RunInfo, monitor *deploytest.ResourceMonitor) error {
-		stackURN, _, _, _, err := monitor.RegisterResource(resource.RootStackType, "test", false)
+		failing, _, _, _, err := monitor.RegisterResource("pkgB:m:typB", "failing", true, deploytest.ResourceOptions{
+			SupportsResultReporting: true,
+		})
 		assert.NoError(t, err)
 
-		val := resource.NewProperty(fmt.Sprintf("step %v", step))
-
-		var outputs resource.PropertyMap
-
-		switch step {
-		case 0, 3:
-			outputs = resource.PropertyMap{
-				"first":  val,
-				"second": val,
-			}
-		case 1:
-			// If an error is raised between calling `pulumi.export("first", ...)` and `pulumi.export("second", ...)`
-			// in SDKs like Python and Go, the first export is still registered via RegisterResourceOutputs.
-			// This test simulates that by not including "second" when the program will error.
-			outputs = resource.PropertyMap{
-				"first": val,
-			}
-		case 2:
-			// The Node.js SDK is a bit different, when an error is thrown between module exports, none of the exports
-			// are included. An empty set of outputs is registered via RegisterResourceOutputs.
-			outputs = resource.PropertyMap{}
-		}
-
-		err = monitor.RegisterResourceOutputs(stackURN, outputs)
+		independent1, _, _, _, err := monitor.RegisterResource(
+			"pkgA:m:typA", "independent1", true, deploytest.ResourceOptions{
+				SupportsResultReporting: true,
```

Comment:
Should we have another test that does the same as this one but without SupportsResultReporting and check the engine still handles that correctly (for old sdks)

Reply:
Good idea, done for this and below :heavy_check_mark: